File: Miscellaneous/Primality.py
#!/bin/python3

import math
import os
import random
import re
import sys
from math import sqrt 

def primality(p):
    if p < 2:
        return 'Not prime'
    elif p <= 3:
        return 'Prime'
    elif p % 2 == 0 or p % 3 == 0:
        return 'Not prime'
    else:
        for i in range(3, int(sqrt(p)+1), 2):
            if p % i == 0:
                return 'Not prime'
    return 'Prime'

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    p = int(input())

    for p_itr in range(p):
        n = int(input())

        result = primality(n)

        fptr.write(result + '\n')

    fptr.close()

File: Miscellaneous/Primality.cpp
#include <bits/stdc++.h>
#include <math.h>

using namespace std;

string primality(int n) {

int c=0;
for(int i=2; i<=sqrt(n);i++)
{
    if(n%i==0){
        c=1;
        break;
    }
    
}
if(c==1||n==1)
    return("Not prime");
else
{return("Prime");}


}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    int p;
    cin >> p;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    for (int p_itr = 0; p_itr < p; p_itr++) {
        int n;
        cin >> n;
        cin.ignore(numeric_limits<streamsize>::max(), '\n');

        string result = primality(n);

        fout << result << "\n";
    }

    fout.close();

    return 0;
}
